### DISCUSSION TITLE
Bencher license

### DISCUSSION THREAD

**Discussion by pkp124** [bencherdev/bencher#275] [General]
Hi all,

I am new to bencher and I am trying to get started in using the tool for my use case. I think its a fantastic tool. This is the closest tool I came across which satisfies my requirements about tracking benchmarks, so thanks for that :) I wanted to play around a bit, before asking anything specific, could you please clarify about your license.

```
All content that resides under any directory or [feature](https://doc.rust-lang.org/cargo/reference/features.html) named "plus" is licensed under the [Bencher Plus License](https://github.com/bencherdev/bencher/blob/main/license/LICENSE-PLUS).
``` 
If I start playing around with self hosted bencher with docker, would I be using any of the "plus" features? What is the intended use case of self hosted? Can we still use self hosted in production envs in terms of license? etc

**Comment by epompeii** [comment]
> I think its a fantastic tool. This is the closest tool I came across which satisfies my requirements about tracking benchmarks, so thanks for that :)

Thank you for the kind words! 😃

> If I start playing around with self hosted bencher with docker, would I be using any of the "plus" features? What is the intended use case of self hosted? Can we still use self hosted in production envs in terms of license? etc

The Bencher License is very much inspired by [the GitLab License](https://gitlab.com/gitlab-org/gitlab/-/blob/master/LICENSE).
You can use Bencher Self-Hosted in dev, test, and production _without_ a Bencher Plus license. The only reason that you need a Bencher Plus license is if you wanted to use some of the Bencher Plus capabilities (ex: private projects, etc), and those require a valid license key to activate.

Does that make sense?
If so, I'll add something like the above to the FAQ on the website.

**Reply by pkp124** [reply]
Makes sense, Thanks for the clarification 🙂 just one more thing, when you say private projects, do you mean private git repositories or projects on bencher platform?
To give you some context, we have some private gitlab c++ projects and I wanted to propose using bencher for continuous benchmarking with self hosted bencher. Should be fine for my usecase?

**Reply by epompeii** [reply]
By "Private Project" I mean on the Bencher platform.

A Public Project is visible to anyone who has access to your Bencher instance.

On Bencher Cloud, [all Public Projects are available here ](https://bencher.dev/perf). If you are using Bencher Self-Hosted, then anyone with access to your network will likewise be able to see all Public Projects. As long as you're okay with that, then it should work just fine for your use case 😃

**Reply by pkp124** [reply]
Great 😃 thanks a lot! I will be coming back with more questions shortly :)

**Comment by pkp124** [comment]
Hi @epompeii 

I have been trying to setup bencher self hosted using docker, just trying it on my local machine. I am stuck at the step of getting self hosted license key from my bencher cloud account. Any suggestions how to proceed?

Thanks

**Reply by epompeii** [reply]
Hey Phani,
I'm very sorry that you're running into issues trying to get Bencher Self-Hosted set up. I've looked into this, and I see what the problem is.

Don't worry, you aren't being charged. I need to fix some things in the backend before I can get you all set up.

Would you prefer for me to update you here or via email?

**Reply by epompeii** [reply]
Just to follow up here.
I got you set up with a free one month, all access license.
The billing issue has now been resolved.

Please, feel free to use that license for the rest of the month 😃
